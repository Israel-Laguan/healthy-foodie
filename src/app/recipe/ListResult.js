import React from "react";
import {
  Row,
  Col,
  Card,
  CardImg,
  CardBody,
  CardTitle,
  CardFooter,
  Badge,
} from "reactstrap";

const styleLinkTo = {
  color: `white`,
  cursor: `pointer`,
};

function ListResult({ items, onLink }) {
  return (
    <>
      {items.length > 0 && (
        <>
          <Row>
            <p>Resultados de la b√∫squeda</p>
          </Row>
          <Row>
            {items.map((item) => (
              <Col md={3} key={`search__recipe-${item.id}`}>
                <Card className="card bg-ligth">
                  <CardImg
                    src={
                      item.image ||
                      "https://img.icons8.com/ios/200/000000/mushbooh-food.png"
                    }
                    top
                    width="100%"
                    alt={item.title}
                    style={styleLinkTo}
                    onClick={() => onLink(item)}
                  />
                  <CardBody
                    className="card-img-overlay"
                    style={styleLinkTo}
                    onClick={() => onLink(item)}
                  >
                    <CardTitle>Proteins: {item.protein}</CardTitle>
                  </CardBody>
                  <CardFooter>
                    <h4 style={styleLinkTo} onClick={() => onLink(item)}>
                      {item.title}
                    </h4>
                    {item.dishTypes.map((dishType) => (
                      <Badge
                        key={`card__footer-${item.id}-${dishType}`}
                        color="success"
                        pill
                      >
                        {dishType}
                      </Badge>
                    ))}
                  </CardFooter>
                </Card>
              </Col>
            ))}
          </Row>
        </>
      )}
    </>
  );
}

export default ListResult;
