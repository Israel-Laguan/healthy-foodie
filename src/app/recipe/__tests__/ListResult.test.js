import React from 'react';
import renderer from 'react-test-renderer';
import ListResult from '../ListResult';

describe('src/app/recipe:details', () => {
  it('render buttons when filters provided', () => {
    const recipes = [
      {
        id: 1,
        image: '',
        healthScore: '23',
        title: '',
        dishTypes: [''],
      },
    ];
    const component = renderer.create(
      <ListResult items={recipes} onLink={e => e.target.value} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('redirect to home when no filters provided', () => {
    const component = renderer.create(
      <ListResult onLink={e => e.target.value} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
