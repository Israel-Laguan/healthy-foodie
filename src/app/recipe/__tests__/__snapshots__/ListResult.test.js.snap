// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/recipe:details redirect to home when no filters provided 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  >
    <div
      className="card bg-ligth card"
    >
      <img
        alt=""
        className="card-img-top"
        onClick={[Function]}
        src="https://img.icons8.com/ios/200/000000/mushbooh-food.png"
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
          }
        }
        width="100%"
      />
      <div
        className="card-img-overlay card-body"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
          }
        }
      >
        <div
          className="card-title"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,50%)",
            }
          }
        >
          Health Score:
           
          <span
            className="badge badge-secondary"
          >
            23
            %
          </span>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <button
          aria-label={null}
          className="text-dark btn btn-link btn-lg btn-block"
          onClick={[Function]}
          type="button"
        >
          
        </button>
        <span
          className="badge badge-success badge-pill"
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/app/recipe:details render buttons when filters provided 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  >
    <div
      className="card bg-ligth card"
    >
      <img
        alt=""
        className="card-img-top"
        onClick={[Function]}
        src="https://img.icons8.com/ios/200/000000/mushbooh-food.png"
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
          }
        }
        width="100%"
      />
      <div
        className="card-img-overlay card-body"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
          }
        }
      >
        <div
          className="card-title"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,50%)",
            }
          }
        >
          Health Score:
           
          <span
            className="badge badge-secondary"
          >
            23
            %
          </span>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <button
          aria-label={null}
          className="text-dark btn btn-link btn-lg btn-block"
          onClick={[Function]}
          type="button"
        >
          
        </button>
        <span
          className="badge badge-success badge-pill"
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;
